src/prpy/serialization.py

```diff
@@ -99,32 +101,51 @@ def serialize_environment_file(env, path, writer=None):
 
     return data
 
-def serialize_kinbody(body):
-    all_joints = []
-    all_joints.extend(body.GetJoints())
-    all_joints.extend(body.GetPassiveJoints())
+def serialize_kinbody(body, uri_only=False):
 
+    uri = body.GetXMLFilename()
+    
+    if uri_only and not uri:
+        serialization_logger.warn(
+            'uri_only passed, but kinbody {} doesn\'t have a URI.'.format(
+            body.GetName()))
```

Comment:
Nit: `kinbody` -> `KinBody`. Also, I would prefer to specifically mention that `GetXMLFilename()` is empty. This will give the user a hint of where to look while debugging this.